### PR TITLE
refactor: consider only tracks with non-zero weight in VertexPerfomanceWriter

### PR DESCRIPTION
**Pull request by felix-russo**
Previously, every track in `TrackAtVertex` contributed to the number of tracks on the reconstructed vertex. This does not make a lot of sense, since some tracks have low weights (which means that they actually do not contribute). This PR excludes low-weight tracks from the histograms to give a better indication of the vertexing performance.

### CODE REVIEW

File: Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.hpp [hunk 1]
```diff
@@ -76,6 +76,8 @@ class VertexPerformanceWriter final
     double truthMatchProbMin = 0.5;
     /// Whether information about tracks is available
     bool useTracks = true;
+    /// minimum track weight for track to be considered as part of the fit
+    double minTrkWeight = 0.1;
```
Comment (paulgessinger) [inline Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.hpp:80]: I guess if we were to default this to 0.1 we would get the exact same output as before?

**Review by paulgessinger — COMMENTED** [review]
Seems reasonable to me. Could you add a few words on the motivation of the change to the PR description?

File: Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.hpp (hunk 1 above)
Comment (felix-russo) [reply Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.hpp:80]: Not quite: We would have to default this to `0.0` to get the same behavior as before.

**Comment by felix-russo** [comment]
> Are the physmon failures expected? `[nTracksRecoVtx](https://acts-herald.app.cern.ch/view/acts-project/acts/1097044433/amvf_seeded.html#nTracksRecoVtx)` goes down a bit, which I guess makes sense?

Yes! Before, every track in `TrackAtVertex` contributed to the number of tracks on the reconstructed vertex. Now, only tracks with weight >= 0.1 contribute; therefore, the number of reconstructed tracks decreases!

File: Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.hpp (hunk 1 above)
Comment (paulgessinger) [reply Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.hpp:80]: Ah yeah, of course.

**Comment by paulgessinger** [comment]
I updated the references, but they might have picked up conflicts.

**Comment by felix-russo** [comment]
CI is green now - can we get this in @paulgessinger?

**Comment by paulgessinger** [comment]
@andiwand cpuld you approve this?

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [ac5823fbf3ea0143acfa670e5327ea60fab31397]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773119)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773118)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773117)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773116)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773115)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773114)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773113)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773112)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773111)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773110)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773109)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773108)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773107)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773106)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773105)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773104)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773103)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773102)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773101)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773100)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773099)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773098)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34773097)||

**Pull request merged**